src/main/java/Duke.java

```diff
@@ -1,10 +1,323 @@
+            } catch (NullCommandArgException | InvalidCommandTimeException
+            | InvalidCommandException | InvalidTaskNumberException e) {
+                printWithBorder(e.getMessage());
+                continue;
+            }
+        }
+    }
+    
+    private static void executeCommand(String commandType, String commandArg) throws NullCommandArgException,
+    InvalidCommandTimeException, InvalidCommandException, InvalidTaskNumberException {
+        switch (commandType) {
+        case "help":
+            printWithBorder(COMMANDS);
+            break;
+        case "list":
+            listAllTasks();
+            break;
+        case "done":
+            markTaskAsDone(commandArg);
+            break;
+        case "todo":
+            addTodo(commandArg);
+            break;
+        case "deadline":
+            addDeadline(commandArg);
+            break;
+        case "event":
+            addEvent(commandArg);
+            break;
+        case "delete":
+            deleteTask(commandArg);
+            break;
+        default:
+            throw new InvalidCommandException(commandType);
+        }
+    }
+
+    private static void listAllTasks() {
+        int count = 1;
+        String listOfTasksString = "Here are the tasks in your list:";
+        for (Task task : taskList) {
+            if (task == null) {
+                break;
+            }
+            listOfTasksString += ("\n    " + Integer.toString(count) + ". " + task.toString());
+            count += 1;
+        }
+        printWithBorder(listOfTasksString);
+    }
+
+    private static void markTaskAsDone(String commandArg) throws NullCommandArgException, InvalidTaskNumberException {
+        checkNullArgs("done", commandArg);
+        int taskNumber = checkNumberValidity(commandArg);
+        
+        Task task = taskList.get(taskNumber - 1);
+        task.setIsDone();
+        String taskSuccessfullyMarkedDoneMessage = "Very nice! I've marked this task as done:\n    " 
+                + task.toString();
+        printWithBorder(taskSuccessfullyMarkedDoneMessage);
+    }
+
+    private static int checkNumberValidity(String commandArg) throws InvalidTaskNumberException {
+        int taskNumber;
+        try {
+            taskNumber = Integer.parseInt(commandArg);
+        } catch (NumberFormatException e) {
+            throw new InvalidTaskNumberException(commandArg);
+        }
+        
+        if (taskNumber < 1 || taskNumber > taskCount) {
+            throw new InvalidTaskNumberException(taskNumber);
+        }
+        return taskNumber;
+    }
+
+    private static void addTodo(String commandArg) throws NullCommandArgException {
+        checkNullArgs("todo", commandArg);
+        Todo task = new Todo(commandArg);
+        addTaskToListAndPrintMessage(task);
+    }
+
+    private static void addDeadline(String commandArg) throws NullCommandArgException, InvalidCommandTimeException {
+        checkNullArgs("deadline", commandArg);
```

Comment:
You could improve on the naming here. Also, If there are unexpected inputs, should the program be allowed execute the lines following this?

Reply:
thank you! i will rename and tidy the methods to improve readability